### PR TITLE
Change instance checks for sympy.Symbol to sympy.Basic

### PR DESCRIPTION
**Pull request by cduck**
For #1930.

Improves support for sympy formulas as gate parameters.  Before this, printing the circuit diagram for circuits containing certain gates with formula parameters would crash.

### CODE REVIEW

File: cirq/google/api/v1/programs.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def _parameterized_value_from_proto_dict(message: Dict
 def _parameterized_value_to_proto_dict(param: Union[sympy.Basic, float]
                                       ) -> Dict:
     out = {}  # type: Dict
-    if isinstance(param, sympy.Symbol):
+    if isinstance(param, sympy.Basic):
```
Comment (dabacon) [inline cirq/google/api/v1/programs.py]: will the result in a formula trying to be passed in to parameter key?  for the proto stuff we only really support string symbols right now.

Comment (cduck) [reply cirq/google/api/v1/programs.py]: Oops.  I was find-replacing too fast.

Comment (cduck) [reply cirq/google/api/v1/programs.py]: Fixed.

File: cirq/ops/phased_x_gate.py [hunk 2]
```diff
@@ -148,7 +148,7 @@ def _circuit_diagram_info_(self, args: protocols.CircuitDiagramInfoArgs
                                ) -> protocols.CircuitDiagramInfo:
         """See `cirq.SupportsCircuitDiagramInfo`."""
 
-        if (isinstance(self.phase_exponent, sympy.Symbol) or
+        if (isinstance(self.phase_exponent, sympy.Basic) or
```
Comment (dabacon) [inline cirq/ops/phased_x_gate.py]: add an example of this to the test_diagram test case.

**Comment by cduck** [comment]
I moved set-dev-version.sh to it's own Travis job because I don't think before deploy will run unless the condition is already met.  (https://docs.travis-ci.com/user/deployment/pypi/#running-commands-before-and-after-release)  This has the benefit that every job won't try push its own dev release (each with a different timestamp).

It seems that `*` characters need to be quoted otherwise Travis doesn't run.

@dabacon PTAL.

**Comment by cduck** [comment]
(Dismissed review for wrong PR)

**Comment by cduck** [comment]
@dabacon PTAL.

File: cirq/optimizers/eject_phased_paulis_test.py [hunk 3]
```diff
@@ -310,28 +315,26 @@ def test_blocked_by_unknown_and_symbols():
...
+        [cirq.X(a)],
+        [cirq.CZ(a, b)**sym],
+        [cirq.X(a)],
+    ),
+                     expected=quick_circuit(
+                         [cirq.X(a)],
+                         [cirq.CZ(a, b)**sym],
+                         [cirq.X(a)],
+                     ),
+                     compare_unitaries=False)
```
Comment (cduck) [inline cirq/optimizers/eject_phased_paulis_test.py]: I'm not sure what yapf did here.

**Pull request merged**